mingwW64: rhash 1.4.0 derivation fails to install

**Describe the bug**
rhash 1.4.0 derivation fails to install using mingwW64 crossSystem.

**To Reproduce**
Try to create a nix-shell with:
```nix
let
  pkgs = import <nixos-unstable> { 
    crossSystem = (import <nixpkgs/lib>).systems.examples.mingwW64;
  };
in
  pkgs.stdenv.mkDerivation {
    name = "test";
    hardeningDisable = [ "all" ]; 

    buildInputs = with pkgs; [
      cmake
    ];
  }
```

**Error message**
```
libtool: install: /nix/store/0y7rv26ffa359wvqd3js94cn73z882fg-coreutils-8.32/bin/install -c .libs/bzip2.exe /nix/store/a1iv7lc3x4mc9hbzcgq1bx7aw8ajx0i6-bzip2-1.0.6.0.1-x86_64-w64-mingw32-bin/bin/bzip2.exe
libtool: install: /nix/store/0y7rv26ffa359wvqd3js94cn73z882fg-coreutils-8.32/bin/install -c .libs/bzip2recover.exe /nix/store/a1iv7lc3x4mc9hbzcgq1bx7aw8ajx0i6-bzip2-1.0.6.0.1-x86_64-w64-mingw32-bin/bin/bzip2recover.exe
make install-binary
 /nix/store/0y7rv26ffa359wvqd3js94cn73z882fg-coreutils-8.32/bin/mkdir -p '/nix/store/a1iv7lc3x4mc9hbzcgq1bx7aw8ajx0i6-bzip2-1.0.6.0.1-x86_64-w64-mingw32-bin/bin'
make[1]: Entering directory '/build/source'
install -d /nix/store/n1ayjj0x0l0fw0z2ggnnm9g9kg91a5ji-rhash-1.4.0-x86_64-w64-mingw32/bin
install -m 755 rhash /nix/store/n1ayjj0x0l0fw0z2ggnnm9g9kg91a5ji-rhash-1.4.0-x86_64-w64-mingw32/bin/rhash
 /nix/store/0y7rv26ffa359wvqd3js94cn73z882fg-coreutils-8.32/bin/install -c bzgrep bzmore bzdiff '/nix/store/a1iv7lc3x4mc9hbzcgq1bx7aw8ajx0i6-bzip2-1.0.6.0.1-x86_64-w64-mingw32-bin/bin'
install: cannot stat 'rhash': No such file or directory
make  install-exec-hook
make[1]: *** [Makefile:75: install-binary] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:68: build-install-binary] Error 2
builder for '/nix/store/1vdn604sz7ld4623kmqpf2hkq4wg5dnb-rhash-1.4.0-x86_64-w64-mingw32.drv' failed with exit code 2
cannot build derivation '/nix/store/b04szmlbz4kj2livkcbh10i7hsvi88d7-cmake-3.19.3-x86_64-w64-mingw32.drv': 1 dependencies couldn't be built
error: build of '/nix/store/b04szmlbz4kj2livkcbh10i7hsvi88d7-cmake-3.19.3-x86_64-w64-mingw32.drv' failed
```

**Expected behavior**
Derivation installs successfully.

**Additional context**
This happens on both stable/unstable channels. Since many packages depend on rhash (for example CMake), this blocks cross compilation for C++ projects and so on.

**Notify maintainers**
@andrewrk 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.15, NixOS, 21.05.git.d73afe9462a (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(bogdb): `"nixos-20.09.3124.2118cf551b9, nixos-unstable-21.05pre269929.ff96a0fa563"`
 - channels(root): `"nixos-20.09.3124.2118cf551b9, nixos-unstable-21.05pre269929.ff96a0fa563"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/tools/security/rhash/default.nix
+++ b/pkgs/tools/security/rhash/default.nix
@@ -34,6 +34,7 @@ stdenv.mkDerivation rec {
 
   configureFlags = [
     "--ar=${stdenv.cc.targetPrefix}ar"
+    "--target=${stdenv.hostPlatform.config}"
     (lib.enableFeature enableStatic "static")
     (lib.enableFeature enableStatic "lib-static")
   ];

